src/Microsoft.OData.Core/UriParser/CustomUriLiteralPrefixes.cs

```diff
@@ -92,12 +92,13 @@ public static bool RemoveCustomLiteralPrefix(string literalPrefix)
         /// </summary>
         /// <param name="literalPrefix">The literal prefix of the EdmType</param>
         /// <returns>Null if the custom literal prefix has no registered EdmType.</returns>
-        internal static IEdmTypeReference GetEdmTypeByCustomLiteralPrefix(string literalPrefix)
+        internal static IEdmTypeReference GetEdmTypeByCustomLiteralPrefix(ReadOnlySpan<char> literalPrefix)
         {
             lock (Locker)
             {
                 IEdmTypeReference edmTypeReference;
-                if (CustomLiteralPrefixesOfEdmTypes.TryGetValue(literalPrefix, out edmTypeReference))
+                string literal = literalPrefix.ToString();
+                if (CustomLiteralPrefixesOfEdmTypes.TryGetValue(literal, out edmTypeReference))

```

Comment:
It sucks that we have to resort to allocations like this when we want to do cache lookups. In .NET 9, we can avoid this with the help of [`IAlternateEqualityComparer`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ialternateequalitycomparer-2?view=net-9.0) which could allow us to look up a dictionary that uses string keys using a `ReadOnlySpan<char>` (see https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-9/#core-collections).

In .NET 8 we can decide whether to settle for the allocations or to implement a workaround if allocation cost is significant.

Reply:
Yep, I should to change to avoid the allocation using .NET 9. In 8 version, current change is a solution because there's no extra allocation happening comparing to the existing code. 
an alternate solution is to create a helper method to compare the `ReadOnlySpan<Char>` and `Dictionary<string, T>`. The method is to traver the dictionary key and get a span of the key, and use the span to compare `ReadOnlySpan<Char>`, It could be a little bit bad perf, I need more investigation.